pkg/protocol/http/http.go

```diff
@@ -123,31 +123,27 @@ func (h *Server) Start(wg *sync.WaitGroup) error {
 		var obj subscriber.Subscriber
 		if err = json.Unmarshal(e.Data(), &obj); err != nil {
 			out.Status = channel.FAILED
-			localmetrics.UpdateSenderCreatedCount(out.Address, localmetrics.FAILED, 1)
 			log.Errorf("failied to parse subscription %s", err)
 		} else {
 			out.Address = obj.GetEndPointURI()
 			if obj.Action == channel.NEW {
 				if _, ok := h.Sender[obj.ClientID]; !ok { // we have a sender object
 					log.Infof("(1)subscriber not found for the following address %s by %s, will attempt to create", e.Source(), obj.GetEndPointURI())
+					out.Status = channel.FAILED
 					if err = h.NewSender(obj.ClientID, obj.GetEndPointURI()); err != nil {
 						log.Errorf("(1)error creating subscriber %v for address %s", err, obj.GetEndPointURI())
-						localmetrics.UpdateSenderCreatedCount(obj.GetEndPointURI(), localmetrics.FAILED, 1)
+					} else if _, err = h.subscriberAPI.CreateSubscription(obj.ClientID, obj); err != nil {
+						log.Infof("failed creating subscription for %s", obj.ClientID.String())
 						out.Status = channel.FAILED
```

Comment:
you dont need this line as you are setting status to failed just before the if/else block. That being said you are better off just setting the status in each if/else condition for better readability